client/retry.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"github.com/cloudwego/kitex/client"
+	"github.com/cloudwego/kitex/pkg/klog"
+	"github.com/cloudwego/kitex/pkg/retry"
+	"github.com/kitex-contrib/config-file/utils"
+)
+
+// WithRetryPolicy returns a server.Option that sets the retry policies for the client
+func WithRetryPolicy(watcher *ConfigWatcher) client.Option {
+	return client.WithRetryContainer(initRetryContainer(watcher))
+}
+
+// initRetryOptions init the retry container
+func initRetryContainer(watcher *ConfigWatcher) *retry.Container {
+	retryContainer := retry.NewRetryContainer()
```

Comment:
Please switch to the `retry.NewRetryContainerWithPercentageLimit()` provided by Kitex v0.7.2 and don't add the close callback for retry container.

You may refer to https://github.com/kitex-contrib/config-nacos/pull/7